tests/test_redis_lock.py

```diff
@@ -499,3 +500,26 @@ def workerfn(unblocked):
     worker.join(0.5)
 
     assert unblocked.value == 1
+
+
+def test_auto_renewal_stops_after_gc(conn):
+    """Auto renewal stops after lock is garbage collected."""
+    lock = Lock(conn, 'spam', auto_renewal=True, expire=1)
+    name = lock._name
+    lock.acquire(blocking=True)
+    lock_renewal_thread = lock._lock_renewal_thread
+    del lock
+    gc.collect()
+
+    slept = 0
+    interval = 0.2
+    while slept <= 10:
+        slept += interval
+        lock_renewal_thread.join(interval)
+        if not lock_renewal_thread.is_alive():
+            break
+
+    time.sleep(2)
+
+    assert not lock_renewal_thread.is_alive()
+    assert conn.get(name) is None
```

Comment:
@ionelmc, Well, that's the best with what I can came up with. And it still doesn't always work on PyPy :astonished: